packages/cli/src/controller/codegen-controller.ts

```diff
@@ -221,27 +221,26 @@ export async function generateAbis(datasources: DatasourceKind[], projectPath: s
   }
 }
 
-function processAbis(sortedAssets: Map<string, string>, projectPath: string): abiRenderProps[] {
+export function processAbis(sortedAssets: Map<string, string>, projectPath: string): abiRenderProps[] {
   const renderInterfaceJobs: abiRenderProps[] = [];
   sortedAssets.forEach((value, key) => {
     const renderProps: abiRenderProps = {name: key, events: [], functions: []};
-    let readAbi = loadFromJsonOrYaml(path.join(projectPath, value)) as abiInterface[];
+    const readAbi = loadFromJsonOrYaml(path.join(projectPath, value)) as abiInterface[] | {abi: abiInterface[]};
     // We need to use for loop instead of map, due to events/function name could be duplicate,
     // because they have different input, and following ether typegen rules, name also changed
     // we need to find duplicates, and update its name rather than just unify them.
 
-    if (!Array.isArray(readAbi)) {
-      readAbi = [readAbi];
-    }
-    const duplicateEventNames = readAbi
+    const abiArray = !Array.isArray(readAbi) && readAbi.abi ? readAbi.abi : (readAbi as unknown as abiInterface[]);
```

Comment:
I think combining the check for array and abi is a bad idea. If the json is an object with out an abi field then we expect `abIArray` to be an array but it wont be